src/go/rdctl/pkg/autostart/autostart_windows.go

```diff
@@ -0,0 +1,84 @@
+package autostart
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sys/windows/registry"
+)
+
+func EnsureAutostart(autostartDesired bool) error {
+	autostartKey, err := registry.OpenKey(registry.CURRENT_USER, `Software\Microsoft\Windows\CurrentVersion\Run`, registry.SET_VALUE)
+	if err != nil {
+		return fmt.Errorf("failed to open registry key: %w", err)
+	}
+	defer autostartKey.Close()
+
+	if autostartDesired {
+		rancherDesktopPath, err := getRancherDesktopPath()
+		if err != nil {
+			return fmt.Errorf("failed to get path to Rancher Desktop: %w", err)
+		}
+		err = autostartKey.SetStringValue("RancherDesktop", rancherDesktopPath)
+		if err != nil {
+			return fmt.Errorf("failed to set RancherDesktop value: %w", err)
```

Comment:
I think a better description would be something like "failed to set Rancher Desktop to start automatically: %w" or something like that.

Reply:
I disagree, but maybe I'm missing something. IMO error messages should describe what the code was trying to do when it failed. So wording like "failed to set Rancher Desktop to start automatically: %w" would be appropriate in the function that is calling `EnsureAutostart` (for example [here](https://github.com/adamkpickering/rancher-desktop/blob/3769-windows-autostart/src/go/rdctl/pkg/factoryreset/delete_data.go#L39)), but not here. In this case, we're trying to modify a registry key, so the error message should convey that.

Though now that I look at it more closely, the error message could be better. I'm going to change it to something more specific. Let me know what you think.